web/data/data.go

```diff
@@ -2,83 +2,58 @@
 package data
 
 import (
-	"fmt"
 	"net/http"
 
 	"github.com/cozy/cozy-stack/couchdb"
+	"github.com/cozy/cozy-stack/web/errors"
 	"github.com/cozy/cozy-stack/web/middlewares"
 	"github.com/gin-gonic/gin"
 )
 
-func abortNoInstance(c *gin.Context) {
-	err := fmt.Errorf("No instance found")
-	c.AbortWithError(http.StatusInternalServerError, err)
-}
-
-func makeCode(err error) int {
-	coucherr, iscoucherr := err.(*couchdb.Error)
-	if iscoucherr {
-		return coucherr.StatusCode
-	}
-
-	return http.StatusInternalServerError
-}
-
-func validDoctype(doctype string) bool {
+func validDoctype(c *gin.Context) {
 	// TODO extends me to verificate characters allowed in db name.
-	return doctype != ""
+	doctype := c.Param("doctype")
+	if doctype == "" {
+		c.AbortWithError(http.StatusBadRequest, errors.InvalidDoctype(doctype))
+	} else {
+		c.Set("doctype", doctype)
+	}
 }
 
 // GetDoc get a doc by its type and id
-func GetDoc(c *gin.Context) {
-	// @TODO this should be extracted to a middleware
-	instance, exists := c.Get("instance")
-	if !exists {
-		abortNoInstance(c)
-		return
-	}
+func getDoc(c *gin.Context) {
+	instance := c.MustGet("instance").(*middlewares.Instance)
+	doctype := c.MustGet("doctype").(string)
```

Comment:
@aenario why do you use `c.MustGet("doctype").(string)` instead of `c.Param("doctype")` here?